js/nodes/materials/structural_material.js

```diff
@@ -0,0 +1,64 @@
+// TODO: Create a material base class
+class StructuralMaterial {
+    constructor()
+    {
+        this.addInput("model_part_name","string");
+        this.properties = {
+            "model_part_name" : "Structure.Parts_Solid_Solid_Auto1",
+            "properties_id"   : 1,
+            "Material"        : {
+                "constitutive_law" : {
+                    "name" : "LinearElasticPlaneStress2DLaw"
+                },
+                "Variables"        : {
+                    "DENSITY"       : 7850.0,
+                    "YOUNG_MODULUS" : 206900000000.0,
+                    "POISSON_RATIO" : 0.29,
+                    "THICKNESS"     : 0.1
```

Comment:
and these would depend on the input above. Just pointing it out as this most probably affects the design.

Reply:
Mm that's interesting, if we want to to this probably we should refactor the whole constitutive law as you say. There is no pretty way to make the variables change dynamically in a node, but we could create an additional node per CL with let the user adjust the variables and their values.

Reply:
I'd go for the second one, create a node per each constitutive law. In my opinion, this would have the next advantages:
1. Keep everything required for a constitutive law to work encapsulated in a single node. This also allow us to set custom defaults for each claw.
2. Makes much easier the addition of new constitutive laws for developers that are not familiar with the flowgraph.